platform/security/idp/security-idp-server/src/main/java/org/codice/ddf/security/idp/binding/api/impl/ValidatorImpl.java

```diff
@@ -0,0 +1,119 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.security.idp.binding.api.impl;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Map;
+
+import org.apache.commons.collections.CollectionUtils;
+import org.apache.commons.collections.Transformer;
+import org.apache.wss4j.common.saml.builder.SAML2Constants;
+import org.codice.ddf.security.idp.binding.api.Validator;
+import org.codice.ddf.security.idp.server.Idp;
+import org.opensaml.saml2.core.AuthnContextClassRef;
+import org.opensaml.saml2.core.AuthnContextComparisonTypeEnumeration;
+import org.opensaml.saml2.core.AuthnRequest;
+import org.opensaml.saml2.metadata.EntityDescriptor;
+import org.opensaml.xml.validation.ValidationException;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import ddf.security.samlp.SimpleSign;
+import ddf.security.samlp.SystemCrypto;
+
+public abstract class ValidatorImpl implements Validator {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ValidatorImpl.class);
+
+    protected final SystemCrypto systemCrypto;
+
+    protected SimpleSign simpleSign;
+
+    private Map<String, EntityDescriptor> serviceProviders;
+
+    public ValidatorImpl(SystemCrypto systemCrypto, Map<String, EntityDescriptor> serviceProviders) {
+        this.systemCrypto = systemCrypto;
+        this.serviceProviders = serviceProviders;
+        this.simpleSign = new SimpleSign(systemCrypto);
+    }
+
+    @Override
+    public void validateAuthnRequest(AuthnRequest authnRequest, String samlRequest,
+            String relayState, String signatureAlgorithm, String signature, boolean strictSignature)
+            throws SimpleSign.SignatureException, ValidationException {
+        if (strictSignature && authnRequest.getAssertionConsumerServiceURL() != null && (
+                authnRequest.getSignature() == null && signature == null)) {
+            throw new IllegalArgumentException(
+                    "Invalid AuthnRequest, defined an AssertionConsumerServiceURL, but contained no identifying signature.");
+        }
+
+        if (authnRequest.getRequestedAuthnContext() != null) {
+            Collection authNContextClasses = CollectionUtils.transformedCollection(
```

Comment:
I'm just going to leave it the way it is for now since we're having issues supporting Java 8 constructs with checkstyle at the moment.